### PR TITLE
kuzu 0.4.1 (new formula)

### PR DESCRIPTION
**Pull request by mewim**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This PR adds [kuzu](https://github.com/kuzudb/kuzu) 0.4.1. [kuzu](https://github.com/kuzudb/kuzu) is an embeddable property graph database management system which implements the Cypher query language.

### CODE REVIEW

File: Formula/k/kuzu.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    args = %w[
+      -DAUTO_UPDATE_GRAMMAR=0
+      -DENABLE_LTO=1
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    bin.install bin/"kuzu_shell" => "kuzu"
```
Comment (chenrui333) [inline Formula/k/kuzu.rb:22]: I think it should be technically symlink?

also, maybe change it in the cmakefile to create artifact as kuzu?

Comment (mewim) [reply Formula/k/kuzu.rb]: Thanks for your review. I have changed it to symlink. We will consider renaming it in cmakefile for our next release.

Comment (SMillerDev) [reply Formula/k/kuzu.rb]: Sounds good, let us know when we can re-review.

Comment (mewim) [reply Formula/k/kuzu.rb]: I think it is ready for re-review now. Our next stable release will be months away. Could we keep the `install_symlink` for now? I will remove this when I bump this formula after the next release.

Comment (chenrui333) [reply Formula/k/kuzu.rb]: any PR or issue for tracking the binary artifact change?

Comment (mewim) [reply Formula/k/kuzu.rb]: OK I created a new issue at our side: https://github.com/kuzudb/kuzu/issues/3458

File: Formula/k/kuzu.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    args = %w[
+      -DAUTO_UPDATE_GRAMMAR=0
+      -DENABLE_LTO=1
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    bin.install_symlink bin/"kuzu_shell" => "kuzu"
```
Comment (chenrui333) [inline Formula/k/kuzu.rb:22]: can you put that issue as code comment into here? Thanks!

Comment (mewim) [reply Formula/k/kuzu.rb]: Done

**Review by chenrui333 — APPROVED** [review]
Thanks!

**Comment by chenrui333** [comment]
@mewim, thanks for your first contribution to homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**